corehq/pillows/tasks.py

```diff
@@ -0,0 +1,32 @@
+from celery.schedules import crontab
+from celery.task import periodic_task
+
+from corehq.apps.es import FormES
+from corehq.form_processor.interfaces.dbaccessors import FormAccessors
+from corehq.form_processor.utils.xform import resave_form
+from corehq.pillows.utils import get_user_type_deep_cache_for_unknown_users
+from corehq.util.datadog.gauges import datadog_gauge
+from corehq.util.decorators import serial_task
+
+
+@periodic_task(run_every=crontab(minute=0, hour=0))
+def fix_user_types():
+    unknown_user_ids = (
+        FormES().user_type('unknown').user_aggregation().run().aggregations.user.keys
+    )
+    datadog_gauge('commcare.fix_user_types.unknown_user_count', len(unknown_user_ids))
+    for user_id in unknown_user_ids:
+        user_type = get_user_type_deep_cache_for_unknown_users(user_id)
+        if user_type != unknown_user_ids:
```

Comment:
@dannyroberts I think this should be `if user_type != UNKNOWN_USER_TYPE`

Reply:
Oh wow, well that explains why this had a slightly larger scope on prod I thought it would. (No serious impact, it just resaved 2-3x as many forms as necessary.)

Reply:
https://github.com/dimagi/commcare-hq/pull/26190